unpause/unpause.go

```diff
@@ -0,0 +1,129 @@
+package unpause
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+	"github.com/jmhodges/clock"
+)
+
+const (
+	// API
+	APIVersion = "v1"
+	APIPrefix  = "/sfe/" + APIVersion
+	GetForm    = APIPrefix + "/unpause"
+
+	// JWT
+	defaultIssuer   = "WFE"
+	defaultAudience = "SFE Unpause"
+)
+
+// JWTClaims represents the claims of a JWT token issued by the WFE for
+// redemption by the SFE. The JWT contains the following claims required for
+// unpausing:
+//   - Subject: the account ID of the Subscriber,
+//   - Version: the API version this JWT was created for,
+//   - Identifiers: a set of paused identifiers.
+type JWTClaims struct {
+	jwt.Claims
+
+	// Version is the API version this JWT was created for.
+	Version string `json:"version"`
+
+	// Identifiers is set of comma separated ACME identifiers.
+	Identifiers string `json:"identifiers"`
+}
+
+// GenerateJWT generates a serialized unpause JWT with the provided claims.
+func GenerateJWT(key []byte, regID int64, identifiers []string, lifetime time.Duration, clk clock.Clock) (string, error) {
+	signer, err := jose.NewSigner(jose.SigningKey{Algorithm: jose.HS256, Key: key}, nil)
+	if err != nil {
+		return "", fmt.Errorf("creating signer: %s", err)
+	}
+
+	claims := JWTClaims{
+		Claims: jwt.Claims{
+			Issuer:    defaultIssuer,
+			Subject:   fmt.Sprintf("%d", regID),
+			Audience:  jwt.Audience{defaultAudience},
+			Expiry:    jwt.NewNumericDate(clk.Now().Add(lifetime)),
+			NotBefore: jwt.NewNumericDate(clk.Now()),
+			IssuedAt:  jwt.NewNumericDate(clk.Now()),
```

Comment:
IssuedAt and NotBefore aren't used in our validation, so there's no reason to include them. They only make the JWT bigger.